sdk/storage/Azure.Storage.Files.Shares/perf/Azure.Storage.Files.Shares.Perf/Scenarios/UploadFile.cs

```diff
@@ -0,0 +1,177 @@
+        /// after the execution of all test instances.
+        /// </summary>
+        public override async Task GlobalCleanupAsync()
+        {
+            await base.GlobalCleanupAsync();
+
+            await FilesShareClient.DeleteAsync();
+        }
+
+        /// <summary>
+        /// Performs the tasks needed to initialize and set up the environment for an instance of the test scenario.
+        /// When multiple instances are run in parallel, setup will be run once for each prior to its execution.
+        /// </summary>
+        public override async Task SetupAsync()
+        {
+            await base.SetupAsync();
+
+            // Create a temporary file and fill it with random characters.
+            // The test will use 'ShareClient' to upload this file.
+            UploadFilePath = Path.GetTempFileName();
+
+            using FileStream uploadFileStream = File.OpenWrite(UploadFilePath);
+            using Stream randomStream = RandomStream.Create(Options.Size);
+
+            await randomStream.CopyToAsync(uploadFileStream);
+
+#if DEBUG
+            Console.WriteLine($"Created local file {UploadFilePath}. Length: {uploadFileStream.Length}");
+#endif
+        }
+
+        /// <summary>
+        /// Performs the tasks needed to clean up the environment for an instance of the test scenario.
+        /// When multiple instances are run in parallel, cleanup will be run once for each after execution has completed.
+        /// </summary>
+        public override async Task CleanupAsync()
+        {
+            await base.CleanupAsync();
+
+            try
+            {
+                File.Delete(UploadFilePath);
+            }
+            catch
+            {
+            }
+        }
+
+        /// <summary>
+        /// Executes the performance test scenario synchronously.
+        /// </summary>
+        /// <param name="cancellationToken">The token used to signal when cancellation is requested.</param>
+        public override void Run(CancellationToken cancellationToken)
+        {
+            DirectoryClientName = Path.GetRandomFileName();
+
+            ShareDirectoryClient directoryClient = FilesShareClient.GetDirectoryClient(DirectoryClientName);
+            directoryClient.Create(cancellationToken: cancellationToken);
+
+            FileClientName = Path.GetFileName(UploadFilePath);
+
+            ShareFileClient fileClient = directoryClient.GetFileClient(FileClientName);
+
+            using (FileStream stream = File.OpenRead(UploadFilePath))
+            {
+                fileClient.Create(stream.Length, cancellationToken: cancellationToken);
+
+                Models.ShareFileUploadInfo fileUploadInfo = fileClient.Upload(stream, cancellationToken: cancellationToken);
+
+#if DEBUG
+                Console.WriteLine($"Uploaded file to {fileClient.Path}. Hash: {fileUploadInfo.ContentHash.Length}");
+#endif
+            }
+        }
```

Comment:
Is it intentional that we are testing both the creation of a directory and the uploading of a file for every file? I feel like this would significantly affect performance readings for smaller files. I'm not intimately familiar with the broader framework, though, so I could be missing something.

Reply:
I changed the code so we are not creating Directory on the storage during test run.